src/Testcontainers.Oracle/OracleBuilder.cs

```diff
@@ -63,6 +77,18 @@ public OracleBuilder WithPassword(string password)
     public override OracleContainer Build()
     {
         Validate();
+
+        var defaultServiceName = GetDefaultServiceName();
+        if (DockerResourceConfiguration.Database == null)
+        {
+            return new OracleContainer(WithDatabase(defaultServiceName).DockerResourceConfiguration);
+        }
+
+        if (DockerResourceConfiguration.Database != defaultServiceName)
+        {
+            return new OracleContainer(WithEnvironment("ORACLE_DATABASE", DockerResourceConfiguration.Database).DockerResourceConfiguration);
+        }
```

Comment:
Doesn't it make sense to set the environment variable in `WithDatabase(string)`? Images that don't support it will ignore it, and we can remove the check here, right?

Reply:
Yes, good point! Addressed in e17bd73ce6a9969e64aef44223e5426222a06546.

Reply:
😕 looks like the container does not properly start if the env var corresponds to the default value… let's revert it, and keep it as it was - sorry.

Reply:
Indeed, reverted in 9af66e27732650c6a8f5f7f5991f2b16f1c9d071. I think that was the exact reason why I wrote it that way and also why I wrote a test that uses the default database name explicitly! 

But this raised another interesting issue: the container failing to start did not end up failing the wait strategy (at least on my machine, using [OrbStack](https://orbstack.dev)). I'll investigate if it's an Oracle specific issue or not.

Reply:
> I'll investigate if it's an Oracle specific issue or not.

It's definitely not an Oracle specific issue. If for some reason a container exits without completing the readiness check then it will wait for the wait strategy timeout (which defaults to 1 hour). I will file a new issue about this.

Reply:
This likely depends on the wait strategy. For example, getting the logs from a stopped container works just fine. That's probably why we try it again and again.